modules/siddhi-core/src/main/java/io/siddhi/core/table/Table.java

```diff
@@ -129,47 +144,99 @@ public TableDefinition getTableDefinition() {
         return tableDefinition;
     }
 
-    public void addEvents(ComplexEventChunk<StreamEvent> addingEventChunk, int noOfEvents) {
-        if (isConnected.get()) {
-            try {
-                if (latencyTrackerInsert != null &&
-                        Level.BASIC.compareTo(siddhiAppContext.getRootMetricsLevel()) <= 0) {
-                    latencyTrackerInsert.markIn();
-                }
-                addingEventChunk.reset();
-                add(addingEventChunk);
-                if (throughputTrackerInsert != null &&
-                        Level.BASIC.compareTo(siddhiAppContext.getRootMetricsLevel()) <= 0) {
-                    throughputTrackerInsert.eventsIn(noOfEvents);
-                }
-            } catch (ConnectionUnavailableException e) {
-                isConnected.set(false);
-                LOG.error(ExceptionUtil.getMessageWithContext(e, siddhiAppContext) +
-                        " Connection unavailable at Table '" + tableDefinition.getId() +
-                        "', will retry connection immediately.", e);
-                connectWithRetry();
-                addEvents(addingEventChunk, noOfEvents);
-            } finally {
-                if (latencyTrackerInsert != null &&
-                        Level.BASIC.compareTo(siddhiAppContext.getRootMetricsLevel()) <= 0) {
-                    latencyTrackerInsert.markOut();
+    public void onAddError(ComplexEventChunk<StreamEvent> addingEventChunk, ConnectionUnavailableException e,
+                           ErrorOccurrence errorOccurrence) {
+        OnErrorAction errorAction = onErrorAction;
+        isConnected.set(false);
+        try {
+            switch (errorAction) {
+                case STORE:
```

Comment:
Let's remove `OnErrorAction errorAction = onErrorAction;` . 
We can use value of `onErrorAction` as follows:
```
        isConnected.set(false);
        try {
            switch (onErrorAction) {
                case STORE:
        ...
```

Let's do the same in other similar methods: `onDeleteError`, `onUpdateError` and `onUpdateOrAddError` as well.

Reply:
Without converting it to a local variable we can't access the enum values https://github.com/siddhi-io/siddhi/blob/1e160212554bf01fb4775d245c4defcca1247a02/modules/siddhi-core/src/main/java/io/siddhi/core/stream/output/sink/Sink.java#L358